src/context/Post.ts

```diff
@@ -330,6 +333,7 @@ export class Data {
         epkNonce: number = 0,
         minRep = 0
     ) {
+        let metadata: Metadata = { id: postId }
```

Comment:
I don't like this shared `metadata` variable. I think it's better to construct it where needed (line 376 and 384). Instead of overwriting properties you can just return an object.

376: ``return { id: `${postId}#${transaction}`, transaction }``
384: `metadata: { id: postId }`